app/src/main/java/com/arctouch/codechallenge/util/NetworkObserver.java

```diff
@@ -0,0 +1,17 @@
+package com.arctouch.codechallenge.util;
+
+import android.app.Application;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+public class NetworkObserver {
+
+    public boolean isNetworkAvailable(Application context) {
```

Comment:
Why does this method receive a parameter of type `Application` specifically?

Reply:
In fact it was not necessary, I should have used the class `Context` as the use of ths parameter is to call the `getSystemService(Context.CONNECTIVITY_SERVICE)` method and this method can be called directly from a instance of `Context` and not necessarily from `Application`.